app/helpers/application_helper.rb

```diff
@@ -364,4 +364,17 @@ def has_out_of_date_configuration?(service)
     return unless service
     service.pending_affecting_changes?
   end
+
+  def rails_asset_url_tag
+    javascript_tag("window.rails_asset_host = '#{rails_asset_host_url}'", { type: "text/javascript"})
+  end
+
+  def rails_asset_host_url
+    asset_host_enabled = Rails.configuration.asset_host.present?
+    asset_host_url = Rails.configuration.three_scale.asset_host.presence
```

Comment:
It is still not clear for me why we are using both settings...
`Rails.configuration.asset_host` and `Rails.configuration.three_scale.asset_host`
Shouldn't we use just one?... 

I guess `Rails.configuration.asset_host` is the one that should define the actual value, as it's env-specific, and will use the value in `Rails.configuration.three_scale.asset_host`, if present.

Reply:
The relevant piece of code is here: [/config/environments/production.rb#L109-L121](https://github.com/3scale/porta/blob/master/config/environments/production.rb#L109-L121)

We could just set `Rails.configuration.asset_host` to `Rails.configuration.three_scale.asset_host` and it would work. The first was converted into a lambda function [here](https://github.com/3scale/system/commit/6043bb025f258c318e46604409b929d7d4943781). I can't get too much info from the commit message: what other assets besides ours are we loading through `javascript_include_tag`? Maybe we can remove the lambda function.

Reply:
Hm, I am not sure about the reason for having this function there.
What I mean is: do you think we can we just rely on `Rails.configuration.asset_host` in this helper?

Reply:
We need both variables, `asset_host_enabled` tells us when the CDN is enabled, it can be disabled when `DISABLE_DIGEST == 1`. In that case we don't care about the value of `RAILS_ASSETS`: all assets loaded by rails won't include the CDN url, so the ones loaded by webpack shouldn't neither.

It's also possible that `asset_host_enabled` is true, but there's not provided asset url (`asset_host_url = RAILS_ASSET_HOST = <empty>`), then the situation is the same, rails won't load any asset and webpack shouldn't neither. So both variables must be true to go ahead and return the CDN url for webpack.

If we rely only on the value of `asset_host_enabled`, then we need to check for the value of `asset_host_url` anyway, to decide if we prepend the` https://` prefix or not. Now it's done in a way all possibilities are checked in one line:

| asset_host_enabled | asset_host_url | return value |
|:---:|:---:|:---:|
| false | '' | '' |
| false  | 'whtaever.cloudfront.net' | ''|
| true  | '' | '' |
| true  | 'whtaever.cloudfront.net' | 'https://whtaever.cloudfront.net' |

Reply:
In this case, does it make sense to rely only on `asset_host_url`, if it is set, then prepend. WDYT? Or is `asset_host_enabled` a rails thing that is tricky to ignore?

Reply:
If we rely only in `asset_host_url`, the second row in the table would fail, it would return `'https://whtaever.cloudfront.net/'` instead of `''`. Since `asset_host_enabled == false`, it will prepend the URL to webpack assets like fonts and JS files, but it won't use the CDN for all other assets. I think that's not the expected behaviour, if the admin disables the CDN, then it must be disabled for all assets.

Reply:
If admin wants to disable CDN, admin could empty `asset_host_url`. Or there is some other issue related to that?

Reply:
It can be disabled in two ways: by `DISABLE_DIGEST == 1` or by `RAILS_ASSET_HOST == ''`. If the admin disables the CDN by leaving `RAILS_ASSET_HOST` empty, that will work, but if they want to enable it solely by `RAILS_ASSET_HOST` it might not work because `DISABLE_DIGEST` could be 1, and then it would only enable the CDN for webpack but not for Sprockets. We need to enforce that no matter how the assets are disabled or enabled, they must be disabled or enabled for both webpack and sprockets.